seirmo/plots/_plot_from_numpy.py

```diff
@@ -0,0 +1,154 @@
+#
+# This file is part of seirmo (https://github.com/SABS-R3-Epidemiology/seirmo/)
+# which is released under the BSD 3-clause license. See accompanying LICENSE.md
+# for copyright notice and full license details.
+#
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class ConfigurablePlotter:
+    def __init__(self):
+        pass
+
+    def begin(self, subplots_rows: int = 1, subplots_columns: int = 1):
+        """
+        Begins creating a figure, with given number of subfigures
+        Replaces init class so object can be reused"""
+        if type(subplots_rows) != int:
+            raise TypeError("Number of rows of subplots must be an integer")
+        if type(subplots_columns) != int:
+            raise TypeError("Number of columns of subplots must be an integer")
+        if subplots_rows <= 0:
+            raise ValueError("Number of rows of subplots must be positive")
+        if subplots_columns <= 0:
+            raise ValueError("Number of columns of subplots must be positive")
+
+        self._fig, self._axes = plt.subplots(subplots_rows, subplots_columns)
+        self._size = subplots_columns * subplots_rows
+        self._nrows = subplots_rows
+        self._ncolumns = subplots_columns
+        # we store a figure object and multiple axes objects
+
+        # Ensure self._axes array is always 2D
+        if self._nrows == 1 and self._ncolumns == 1:
+            self._axes = np.array(self._axes)[np.newaxis, np.newaxis]
+        elif self._nrows == 1:
+            self._axes = np.array(self._axes)[np.newaxis, :]
+        elif self._ncolumns == 1:
+            self._axes = np.array(self._axes)[:, np.newaxis]
+
+    def __getitem__(self, index):
+        """If figure = ConfigurablePlotter(), then figure.begin().
+        Figure[0] will return the matplot figure, and figure[1] will
+        return the subplot axis objects"""
+        if index == 0:
+            return self._fig
+        elif index == 1:
+            return self._axes
+        else:
+            raise ValueError("Index must be 0 (for figure) or 1 (for axes)")
+
+    def add_data_to_plot(
+        self,
+        times: np.ndarray,
+        data_array: np.ndarray,
+        position: list = [0, 0],
+        xlabel: str = "time",
+        ylabels: list = [],
```

Comment:
If a string is passed to ylabels (which you do in one of the notebooks), the behavior is undesirable in general.
If strings are allowed, they should be interpreted as the label in the case data_width=1

Reply:
Good catch on the colors as well

Reply:
So the notebook was an error - the documentation says it _should_ be a list! However if I made that error then the user could too, so I now handle it explicitly, converting the string into a single element list. I've updated the documentation appropriately, and added similar tests with colour with more comprehensive testing - apologies for missing that the first time around!